modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4MessageChannelHandler.java

```diff
@@ -166,18 +159,29 @@ private void doFlush(ChannelHandlerContext ctx) {
                     }
                 });
             }
-            ctx.flush();
-            if (channel.isActive() == false) {
```

Comment:
This means we're no longer checking `channel.isActive()` in the loop, could we end up stuck and/or trying to write to a closed channel?

Reply:
No that should be fine, the writes to the closed channel would just fail. As long as we check after the loop to make sure "not writable" wasn't the result of the channel being closed we're good I think.